src/poetry/console/commands/self/self_command.py

```diff
@@ -57,7 +57,7 @@ def default_group(self) -> str:
     def activated_groups(self) -> set[str]:
         return {self.default_group}
 
-    def generate_system_pyproject(self) -> None:
+    def generate_system_pyproject(self) -> str:
```

Comment:
This was a refactor to support generating the system pyproject.toml and modifying it in memory that turned out not to be true. I left the refactor because it felt cleaner afterward.